.github/workflows/test_verification.yml

```diff
@@ -0,0 +1,144 @@
+name: Tests
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  Linux_gcc:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+
+        gccver: [ 7, 8, 9, 10 ]
+    steps:
+      - name: Set up cmake
+        uses: jwlawson/actions-setup-cmake@v1.11
+        with:
+          cmake-version: '3.11.x'
+
+      - name: Set up gcc
+        uses: egor-tensin/setup-gcc@v1
+        with:
+          version: ${{ matrix.gccver }}
+          platform: x64
+
+      - uses: actions/checkout@v2
+
+      - name: Make build folder
+        working-directory: ${{github.workspace}}
+        run: mkdir build
+
+      - name: CMake
+        working-directory: ${{github.workspace}}/build
+        run: cmake ../
+
+      - name: Build
+        working-directory: ${{github.workspace}}/build
+        run: cmake --build . -- -j4
+
+      - name: Test
+        working-directory: ${{github.workspace}}/build/tests/
+        run: ./tileson_tests --order rand
+
+  Linux_clang:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        clangver: [ 8, 11 ]
+    steps:
+      - name: Set up cmake
+        uses: jwlawson/actions-setup-cmake@v1.11
+        with:
+          cmake-version: '3.11.x'
+
+      - name: Set up clang
+        uses: egor-tensin/setup-clang@v1
+        with:
+          version: ${{ matrix.clangver }}
+          platform: x64
+
+      - uses: actions/checkout@v2
+
+      - name: Make build folder
+        working-directory: ${{github.workspace}}
+        run: mkdir build
+
+      - name: CMake
+        working-directory: ${{github.workspace}}/build
+        run: cmake ../
+
+      - name: Build
+        working-directory: ${{github.workspace}}/build
+        run: cmake --build . -- -j4
+
+      - name: Set Swap Space
+        uses: pierotofy/set-swap-space@master
+        with:
+          swap-size-gb: 2
+
+      - name: Test
+        working-directory: ${{github.workspace}}/build/tests/
+        run: ./tileson_tests --order rand
+
+
+  Windows:
+    runs-on: windows-2019
+    steps:
+      - name: Add msbuild to PATH
+        uses: microsoft/setup-msbuild@v1.1
+
+      - name: Set up cmake
+        uses: jwlawson/actions-setup-cmake@v1.11
+        with:
+          cmake-version: '3.11.x'
+
+      - uses: actions/checkout@v2
+
+      - name: Make build folder
+        working-directory: ${{github.workspace}}
+        run: mkdir build
+
+      - name: CMake
+        working-directory: ${{github.workspace}}/build
+        run: cmake ../ -G "Visual Studio 15 2017 Win64"
+
+      - name: Build
+        working-directory: ${{github.workspace}}/build
+        run: cmake --build .
+
+      - name: Test
+        working-directory: ${{github.workspace}}/build
+        run: tests/Debug/tileson_tests.exe --order rand
+
+#  Mac_clang:
```

Comment:
osx is currently commented out, as of clang++-12/clang++-13 issues. I do not know how to install specific clang versions on osx.

Reply:
I created a project some while ago to test Github Actions, where I also struggled to get OSX to fork with Clang, but `GCC` with OSX worked fine. I guess Apple is just Apple, and have done some weird changes with Clang and how it locates standard libraries. For reference, that project can be found here:
https://github.com/SSBMTonberry/github-actions-tests

Reply:
Oh, and it's fine that OSX is not included for now. Anything is better than nothing after all 🙂